pkg/api/controller.go

```diff
@@ -4286,6 +4286,94 @@ func (c *Controller) HeadObject(w http.ResponseWriter, r *http.Request, reposito
 	}
 }
 
+func (c *Controller) GetMetadataObject(w http.ResponseWriter, r *http.Request, repository string, objectType string, objectID string, params apigen.GetMetadataObjectParams) {
+	const getTypeMetaRange = "meta_range"
+	const getTypeRange = "range"
+
+	if !c.authorize(w, r, permissions.Node{
+		Type: permissions.NodeTypeAnd,
+		Nodes: []permissions.Node{
+			{
+				Permission: permissions.Permission{
+					Action:   permissions.ListObjectsAction,
+					Resource: permissions.RepoArn(repository),
+				},
+			},
+			{
+				Permission: permissions.Permission{
+					Action:   permissions.ReadRepositoryAction,
+					Resource: permissions.RepoArn(repository),
+				},
+			},
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "get_metadata_object", r, repository, "", "")
+
+	repo, err := c.Catalog.GetRepository(ctx, repository)
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+
+	var objPath string
+	if objectType == getTypeMetaRange {
+		addr, err := c.Catalog.GetMetaRange(ctx, repository, objectID)
+		if c.handleAPIError(ctx, w, r, err) {
+			return
+		}
+		objPath = string(addr)
+	} else if objectType == getTypeRange {
+		addr, err := c.Catalog.GetRange(ctx, repository, objectID)
```

Comment:
Both `c.Catalog.GetMetaRange`  and `c.Catalog.GetRange` return the SSTable of the `objectID` regardless of it's type. 
i.e I can call `http://localhost:8000/api/v1/repositories/test/metadata/object/meta_range/<range-id>` or `http://localhost:8000/api/v1/repositories/test/metadata/object/range/<meta-range_id>`. 

Is this a bug in general? I'm not sure. 
But, I wonder if it's a necessary to have the the enum `type` in the request at all? Since we can just call one of them.

Reply:
both catalog methods don't check for existence _or_ type of returned objects (if you drill into the 7 layers of abstractions below the catalog, this is essentially a path.Join() on the storage namespace with the ID of the requested file.

I think that's fine. If we ever decide to use a different format, or to seperate meta-ranges from ranges into sub-directories (and thus change what `Catalog` is returning), we won't have to introduce a change here. So better be forward compatible I guess...